frame/bounties/src/lib.rs

```diff
@@ -1429,12 +1445,23 @@ impl<T: Config> Module<T> {
 
 	fn ensure_bounty_active(
 		bounty_id: BountyIndex,
+		is_reject_origin: bool,
 	) -> Result<(T::AccountId, T::BlockNumber), DispatchError> {
+
 		let bounty = Self::bounties(&bounty_id).ok_or(Error::<T>::InvalidIndex)?;
+
 		if let BountyStatus::Active { curator, update_due } = bounty.status {
 			Ok((curator, update_due))
 		} else {
-			Err(Error::<T>::UnexpectedStatus.into())
+			if is_reject_origin {
+				// If call originates from T::RejectOrigin
+				// state check on parent bounty is ignored,
+				// and execution is forced with dummy object
+				// This object is not to get used by the caller.
+				Ok((Default::default(), Default::default()))
```

Comment:
I think the function should be renamed `ensure_bounty_exist` and return type with Option:
either `Option<(T::AccountId, T::BlockNumber)>` or `(Option<T::AccountId>, Option<T::BlockNumber>)`

Using default here is a bit missleading IMO

Reply:
Sure :+1:. Modified the function as suggested, Please let me know if any deviation.